### PR TITLE
add marker parameter for rds3 describe_db_snapshots

### PR DESCRIPTION
**Pull request by harshad-aws**
This PR fixes issue with the moto's rds3 module which doesn't support marker keyword argument.

### CODE REVIEW

File: moto/rds3/models/db_snapshot.py [hunk 1]
```diff
@@ -157,5 +157,24 @@ def describe_db_snapshots(self, db_instance_identifier=None, db_snapshot_identif
                 if snapshot.db_instance_identifier == db_instance_identifier:
                     if snapshot.snapshot_type in snapshot_types:
                         db_instance_snapshots.append(snapshot)
-            return db_instance_snapshots
-        return self.db_snapshots.values()
+            if marker:
+                start = db_instance_snapshots.index(marker) + 1
```
Comment (mlsanjaysrivatsa) [inline moto/rds3/models/db_snapshot.py:161]: what if the marker is out of the bounds of the index ?

File: moto/rds3/models/db_snapshot.py [hunk 2]
```diff
@@ -157,5 +157,24 @@ def describe_db_snapshots(self, db_instance_identifier=None, db_snapshot_identif
...
+        if marker:
+            start = all_db_snapshots.index(marker) + 1
+        else:
+            start = 0
+        page_size = max_records if max_records else 100
+        db_snapshots_resp = all_db_snapshots[start : start + page_size]
+        next_marker = None
+        if len(all_db_snapshots) > start + page_size:
+            next_marker = db_snapshots_resp[-1].db_snapshot_identifier
+        return (db_snapshots_resp, next_marker)
```
Comment (mlsanjaysrivatsa) [inline moto/rds3/models/db_snapshot.py:180]: can you see if you can avoid duplication of code

File: moto/rds3/models/db_snapshot.py [hunk 3]
```diff
@@ -147,7 +147,7 @@ def delete_db_snapshot(self, db_snapshot_identifier):
         snapshot.delete_events()
         return self.db_snapshots.pop(db_snapshot_identifier)
 
-    def describe_db_snapshots(self, db_instance_identifier=None, db_snapshot_identifier=None, snapshot_type=None):
+    def describe_db_snapshots(self, db_instance_identifier=None, db_snapshot_identifier=None, snapshot_type=None, marker=None, max_records=None):
```
Comment (mgshirali) [inline moto/rds3/models/db_snapshot.py:150]: Handle input validation for max_records
Also, should this closely mimic aws cli which allows between 20-100?
```
MaxRecords. Must be between 20 and 100
```

File: moto/rds3/models/db_snapshot.py (hunk 1 above)
Comment (mgshirali) [reply moto/rds3/models/db_snapshot.py:161]: Also, what if db_instance_snapshots[] is empty. Add a check?

**Review by mgshirali — COMMENTED** [review]
Provided comments

File: moto/rds3/models/db_snapshot.py (hunk 1 above)
Comment (harshad-aws) [reply moto/rds3/models/db_snapshot.py]: >Marker is an optional pagination token provided by a previous request. If this parameter is specified, the response includes only records beyond the marker, up to the value specified by MaxRecords. 

As quoted in this, marker generally is not set by user and used for pagination purpose.

**Comment by mlsanjaysrivatsa** [comment]
https://github.com/harshad-aws/moto/blob/rds3-for-py3/moto/rds3/responses.py#L65

As you can see the responses already factors in marker in parameters. If the describe_db_snapshots call fails in the argument signature just correct the signature and we should be good. @harshad-aws

**Comment by harshad-aws** [comment]
> https://github.com/harshad-aws/moto/blob/rds3-for-py3/moto/rds3/responses.py#L65
> 
> As you can see the responses already factors in marker in parameters. If the describe_db_snapshots call fails in the argument signature just correct the signature and we should be good. @harshad-aws

Have updated method signature as discussed.

**Comment by mlsanjaysrivatsa** [comment]
Looks good to me. Thanks @harshad-aws

**Comment by harshad-aws** [comment]
> @harshad-aws Can you add a paginated test to the `test_db_snapshots.py` file? Will look very similar to the one we already have for cluster snapshots here:
> 
> https://github.com/bpandola/moto/blob/bb161507febd8497225d007f8debf28c1a37ba7a/tests/test_rds3/test_db_cluster_snapshots.py#L40
> 
> 
> Thanks

Hi @bpandola I have added the unit test.

**Comment by bpandola** [comment]
@harshad-aws This is merged.  Use commit hash `1ded070` for `moto` requirement in RDS.

**Pull request merged**